pkg/controlplane/client.go

```diff
@@ -78,16 +90,90 @@ func (c *client) Authorize(req *api.AuthorizationRequest) (*remoteServerAuthoriz
 	return resp, nil
 }
 
+// IsActive returns if the peer is active or not.
+func (c *client) IsActive() bool {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+	return c.active
+}
+
+// setActive the peer status (active or not).
+func (c *client) setActive(active bool) {
+	c.lock.Lock()
+	defer c.lock.Lock()
+	c.active = active
+	if active || c.lastSeen.IsZero() {
+		c.lastSeen = time.Now()
+	}
+}
+
+// GetHeartbeat get a heartbeat from other peers.
+func (c *client) getHeartbeat() error {
+	var retErr error
+	// copy peer clients array aside
+	peerClients := make([]*jsonapi.Client, len(c.clients))
+	{
+		c.lock.RLock()
+		defer c.lock.RUnlock()
+		copy(peerClients, c.clients)
+	}
+
+	for _, client := range peerClients {
+		serverResp, err := client.Get(api.HeartbeatPath)
+		if err != nil {
+			retErr = errors.Join(retErr, err)
+		} else {
+			if serverResp.Status == http.StatusOK {
+				return nil
+			}
+			retErr = errors.Join(retErr, fmt.Errorf("unable to get heartbeat (%d), server returned: %s",
+				serverResp.Status, serverResp.Body))
+		}
+	}
+
+	return retErr // Return an error if all client targets are unreachable
+}
+
+// heartbeatMonitor checks all peers for responsiveness, every fixed amount of time.
+func (c *client) heartbeatMonitor() {
+	c.logger.Info("Starting sending heartbeats to peer")
```

Comment:
might be worthwhile to keep the peer name on the `client` so it can be used for more informative logging message

Reply:
It is part of the logger see line 175